### DISCUSSION TITLE
Transfer stereochemical information from SDF to enhanced stereochem SMILES

### DISCUSSION THREAD

**Discussion by zackterray** [rdkit/rdkit#6636] [Q&A] [answered]
I was wondering if there is a way to transfer stereochemical information from an SDF to enhanced stereochem smiles.  When I try to convertthese mols with `Chem.MolToCXSmiles(m, params=write_params, flags=Chem.rdmolfiles.CXSmilesFields.CX_ENHANCEDSTEREO)`, no additional enhanced stereochemistry fields are written to the racemic example.    
Two examples from enamine:
**Racemic Molecule:**

  Mrv2013 08012310322D          

  9  8  0  0  0  0            999 V2000
    0.0000    0.0000    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
    3.3147    1.3406    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
    3.3147    0.5156    0.0000 C   0  0  2  0  0  0  0  0  0  0  0  0
    4.0291    0.1031    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.0291   -0.7219    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.3147   -1.1344    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.6002   -0.7219    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.6002    0.1031    0.0000 C   0  0  2  0  0  0  0  0  0  0  0  0
    1.8857    0.5156    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
  3  2  1  6  0  0  0
  3  4  1  0  0  0  0
  4  5  1  0  0  0  0
  5  6  1  0  0  0  0
  6  7  1  0  0  0  0
  8  7  1  0  0  0  0
  3  8  1  0  0  0  0
  8  9  1  6  0  0  0
M  END
>  <Mw>
151.63

>  <ID>
EN300-244068

>  <IUPAC Name>
rac-(1R,2S)-2-aminocyclohexan-1-ol hydrochloride

>  <URL>
https://www.enaminestore.com/catalog/EN300-244068

>  <Stock_weight_G>
30

>  <Class>
Alcohols, Primary_Amines

>  <Subclass>
aliphatic_alcohols, PriAmines_Aliphatic

$$$$

**Enantiopure Molecule:**

  Mrv2013 08012310322D          

 10 11  0  0  1  0            999 V2000
    2.5423    1.5837    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.1267    0.8710    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.5362    0.1547    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.1206   -0.5580    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.2957   -0.5544    0.0000 C   0  0  2  0  0  0  0  0  0  0  0  0
    1.2326    0.1301    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.3017    0.8745    0.0000 C   0  0  1  0  0  0  0  0  0  0  0  0
    0.8862    0.1618    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.1594    0.5522    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.1561   -0.2224    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  2  3  2  0  0  0  0
  3  4  1  0  0  0  0
  5  4  1  0  0  0  0
  5  6  1  6  0  0  0
  7  6  1  6  0  0  0
  7  2  1  0  0  0  0
  7  8  1  0  0  0  0
  5  8  1  0  0  0  0
  8  9  1  0  0  0  0
  8 10  1  0  0  0  0
M  END
>  <Mw>
136.23

>  <ID>
EN300-90458

>  <IUPAC Name>
(1R,5R)-2,6,6-trimethylbicyclo[3.1.1]hept-2-ene

>  <URL>
https://www.enaminestore.com/catalog/EN300-90458

>  <Stock_weight_G>
2228.134

>  <Class>
Alkenes

>  <Subclass>
Alkenes

$$$$

**Comment by ricrogz** [comment, marked as answer]
Hi @zackterray,

Your distinction between racemic/enantiopure seems to be based on the chiral flag in the counts line in these mol blocks. The use of this flag, besides being obsolete, is contentious, since it is not very well documented in the CT File Formats standard: most of the people don't care about the chiral flag at all, and others have come up with different meanings particular to their own organizations. For these reasons, RDKit mostly ignores that flag.

In any case, you can convert the chiral flag based enhanced stereochemistry into stereogroups by using the `Chem.TranslateChiralFlagToStereoGroups()` function (check the documentation of the function for usage and available arguments):

```
In [2]: m1 = Chem.MolFromMolBlock(mb1)  # the "racemic molecule"
   ...: m2 = Chem.MolFromMolBlock(mb2)  # the "enantiopure molecule"
   ...: 
   ...: p = Chem.SmilesWriteParams()
   ...: f = Chem.rdmolfiles.CXSmilesFields.CX_ENHANCEDSTEREO
   ...: print(Chem.MolToCXSmiles(m1, p, f))
   ...: print(Chem.MolToCXSmiles(m2, p, f))
Cl.N[C@@H]1CCCC[C@@H]1O
CC1=CC[C@@H]2C[C@H]1C2(C)C

In [3]: Chem.TranslateChiralFlagToStereoGroups(m1)
   ...: Chem.TranslateChiralFlagToStereoGroups(m2)
   ...: 
   ...: print(Chem.MolToCXSmiles(m1, p, f))
   ...: print(Chem.MolToCXSmiles(m2, p, f))
Cl.N[C@H]1CCCC[C@H]1O |&1:2,7|
CC1=CC[C@@H]2C[C@H]1C2(C)C |a:4,6|
```

Once you have run the transformation function, the Enhanced Stereo Groups will stay on the mols, and not only show up on CXSmiles, but also on Mol V3000:

```
In [4]: print(Chem.MolToMolBlock(m1))

     RDKit          2D

  0  0  0  0  0  0  0  0  0  0999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 9 8 0 0 0
M  V30 BEGIN ATOM
M  V30 1 Cl 0.000000 0.000000 0.000000 0
M  V30 2 N 3.314700 1.340600 0.000000 0
M  V30 3 C 3.314700 0.515600 0.000000 0
M  V30 4 C 4.029100 0.103100 0.000000 0
M  V30 5 C 4.029100 -0.721900 0.000000 0
M  V30 6 C 3.314700 -1.134400 0.000000 0
M  V30 7 C 2.600200 -0.721900 0.000000 0
M  V30 8 C 2.600200 0.103100 0.000000 0
M  V30 9 O 1.885700 0.515600 0.000000 0
M  V30 END ATOM
M  V30 BEGIN BOND
M  V30 1 1 3 2 CFG=3
M  V30 2 1 3 4
M  V30 3 1 4 5
M  V30 4 1 5 6
M  V30 5 1 6 7
M  V30 6 1 8 7
M  V30 7 1 3 8
M  V30 8 1 8 9 CFG=3
M  V30 END BOND
M  V30 BEGIN COLLECTION
M  V30 MDLV30/STERAC1 ATOMS=(2 3 8)
M  V30 END COLLECTION
M  V30 END CTAB
M  END
```

I hope this answers your question :)

**Reply by zackterray** [reply]
Yes, that's exactly what I wanted!  Thank you so much for the help and for the lesson on stereochemistry!
